explorer/interpreter/type_checker.cpp

```diff
@@ -2048,17 +2065,19 @@ auto TypeChecker::Substitute(const Bindings& bindings,
 
 auto TypeChecker::RebuildValue(Nonnull<const Value*> value) const
     -> ErrorOr<Nonnull<const Value*>> {
-  return SubstituteImpl(Bindings(), value);
+  return SubstituteImpl(Bindings(), value,
+                        SourceLocation::DiagnosticsIgnored());
```

Comment:
We're not ignoring the diagnostics produced by this function, so we should pass in a real source location here. The only caller appears to be `ConstraintTypeBuilder`, though, and as noted above we can use the location of its self binding as the location of its diagnostics.